src/dpdkstat.c

```diff
@@ -274,105 +288,106 @@ static int dpdk_re_init_shm()
   if (reset) {
     g_configuration->eal_initialized = 0;
     g_configuration->num_ports = 0;
-    memset(&g_configuration->xstats, 0, g_configuration->num_xstats* sizeof(struct rte_eth_xstats));
+    g_configuration->xstats = NULL;
     g_configuration->num_xstats = 0;
     for (int i = 0; i < RTE_MAX_ETHPORTS; i++)
       g_configuration->num_stats_in_port[i] = 0;
   }
   close(g_configuration->helper_pipes[1]);
   int err = kill(g_configuration->helper_pid, SIGKILL);
   if (err) {
-    ERROR("dpdkstat: error sending kill to helper: %s\n", strerror(errno));
+    char errbuf[ERR_BUF_SIZE];
+    WARNING("dpdkstat: error sending kill to helper: %s", sstrerror(errno, errbuf,
+      sizeof (errbuf)));
   }
 
   return 0;
 }
 
 static int dpdk_helper_spawn(enum DPDK_HELPER_ACTION action)
 {
+  char errbuf[ERR_BUF_SIZE];
   g_configuration->eal_initialized = 0;
   g_configuration->helper_action = action;
   /*
    * Create a pipe for helper stdout back to collectd. This is necessary for
    * logging EAL failures, as rte_eal_init() calls rte_panic().
    */
-  if (g_configuration->helper_pipes[1]) {
-    DEBUG("dpdkstat: collectd closing helper pipe %d\n",
-          g_configuration->helper_pipes[1]);
-  } else {
-    DEBUG("dpdkstat: collectd helper pipe %d, not closing\n",
-          g_configuration->helper_pipes[1]);
-  }
-  if(pipe(g_configuration->helper_pipes) != 0) {
-    DEBUG("dpdkstat: Could not create helper pipe: %s\n", strerror(errno));
+  if (pipe(g_configuration->helper_pipes) != 0) {
+    DEBUG("dpdkstat: Could not create helper pipe: %s", sstrerror(errno, errbuf,
+      sizeof (errbuf)));
     return -1;
   }
 
   int pipe0_flags = fcntl(g_configuration->helper_pipes[0], F_GETFL, 0);
   int pipe1_flags = fcntl(g_configuration->helper_pipes[1], F_GETFL, 0);
   if (pipe0_flags == -1 || pipe1_flags == -1) {
-    ERROR("dpdkstat: error setting up pipe flags: %s\n", strerror(errno));
+    WARNING("dpdkstat: Failed setting up pipe flags: %s", sstrerror(errno, errbuf,
+      sizeof (errbuf)));
   }
-  int  pipe0_err = fcntl(g_configuration->helper_pipes[0], F_SETFL, pipe1_flags
-                         | O_NONBLOCK);
-  int  pipe1_err = fcntl(g_configuration->helper_pipes[1], F_SETFL, pipe0_flags
-                         | O_NONBLOCK);
+  int pipe0_err = fcntl(g_configuration->helper_pipes[0], F_SETFL, pipe1_flags
+    | O_NONBLOCK);
+  int pipe1_err = fcntl(g_configuration->helper_pipes[1], F_SETFL, pipe0_flags
+    | O_NONBLOCK);
   if (pipe0_err == -1 || pipe1_err == -1) {
-    ERROR("dpdkstat: error setting up pipes: %s\n", strerror(errno));
+    WARNING("dpdkstat: Failed setting up pipes: %s", sstrerror(errno, errbuf,
+      sizeof (errbuf)));
   }
 
   pid_t pid = fork();
   if (pid > 0) {
     close(g_configuration->helper_pipes[1]);
     g_configuration->helper_pid = pid;
-    DEBUG("dpdkstat: helper pid %u\n", g_configuration->helper_pid);
+    DEBUG("dpdkstat: helper pid %u", g_configuration->helper_pid);
```

Comment:
You're printing a `pid_t` here, which according to *Single UNIX* is a "signed integral type". As far as I know there is no specialized format option for this type, so I'd add an explicit cast to be safe:

```c
DEBUG ("… %ld …", (long) g_configuration->helper_pid);
```

(I'm using `long` here because it is usually set to the processor's word size and therefore a good default for types of unknown size, but `int` should work equally well.)